### PR TITLE
Updated the generic mechanisms to build transactions to the Babbage era.

### PR DESCRIPTION
**Pull request by TimSheard**

### CODE REVIEW

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs [hunk 1]
```diff
@@ -36,6 +37,11 @@ import Cardano.Ledger.Alonzo.Tx (hashScriptIntegrity)
 import qualified Cardano.Ledger.Alonzo.Tx as Alonzo
 import qualified Cardano.Ledger.Alonzo.TxBody as Alonzo (TxOut (..))
 import Cardano.Ledger.Alonzo.TxWitness (Redeemers (..), TxDats (..), TxWitness (..), unTxDats)
+--
+
```
Comment (JaredCorduan) [inline libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs]: I'm surprised ormolu allows this...

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs [hunk 2]
```diff
@@ -619,6 +689,27 @@ updatePParams (Alonzo _) pp dpp = case dpp of
   MaxBlockExUnits n -> pp {Alonzo._maxBlockExUnits = n}
   CollateralPercentage perc -> pp {Alonzo._collateralPercentage = perc}
   _ -> pp
+updatePParams (Babbage _) pp dpp = case dpp of
```
Comment (JaredCorduan) [inline libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs:365]: should we make a comment that we correctly and intentionally are dropping `d` and `extraEntropy` here?

**Review by JaredCorduan — APPROVED** [review]
Looks good to me

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs [hunk 3]
```diff
@@ -245,6 +245,14 @@ data Datum era
   | Datum !(BinaryData era)
   deriving (Eq, Ord, Show)
 
+instance Semigroup (Datum era) where -- This instance is left biased
```
Comment (JaredCorduan) [inline eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs]: was this instance needed?

Comment (TimSheard) [reply eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs]: Yes it is used in merging in the generic stuff.

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs (hunk 2 above)
Comment (TimSheard) [reply libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs:365]: this is supposed to be the union over all Eras, so extraEntropy is still around in past Eras.

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs (hunk 1 above)
Comment (TimSheard) [reply libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Updaters.hs]: its gone now

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Indexed.hs [hunk 4]
```diff
@@ -307,4 +314,4 @@ instance PrettyA (SKey kr c) where
 instance PrettyA (MultiAsset era) where
   prettyA (MultiAsset v) = prettyA v
 
-type Good c = DSignable c (CH.Hash (CC.HASH c) EraIndependentTxBody)
+type Goode c = DSignable c (CH.Hash (CC.HASH c) EraIndependentTxBody)
```
Comment (JaredCorduan) [inline libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Indexed.hs]: goodie goodie gumdrops?

Comment (TimSheard) [reply libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Indexed.hs]: This was a leftover wart. It has been removed.

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs (hunk 3 above)
Comment (TimSheard) [reply eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs]: Not true any more, with the simplified merging. So I removed it.

**Pull request merged**